### PR TITLE
Refactoring to adhere to arch best practices

### PR DESCRIPTION
**Pull request by florina-muntenescu**

### CODE REVIEW

**Comment by florina-muntenescu** [comment]
@manuelvicnt - not sure why I can't find you in the list of reviewers, but this is ready for review

File: app/src/main/java/com/example/android/roomwordssample/WordRepository.kt [hunk 1]
```diff
@@ -26,7 +26,7 @@ class WordRepository(private val wordDao: WordDao) {
 
     // Room executes all queries on a separate thread.
     // Observed LiveData will notify the observer when the data has changed.
```
Comment (manuelvicnt) [inline app/src/main/java/com/example/android/roomwordssample/WordRepository.kt:28]:
```suggestion
    // Observed Flow will notify the observer when the data has changed.
```

File: app/src/main/java/com/example/android/roomwordssample/WordViewModel.kt [hunk 2]
```diff
@@ -28,20 +29,15 @@ import kotlinx.coroutines.launch
...
     // - Repository is completely separated from the UI through the ViewModel.
-    val allWords: LiveData<List<Word>>
-
-    init {
-        val wordsDao = WordRoomDatabase.getDatabase(application, viewModelScope).wordDao()
-        repository = WordRepository(wordsDao)
-        allWords = repository.allWords
-    }
+    val allWords: LiveData<List<Word>> = repository.allWords.asLiveData(
+        viewModelScope.coroutineContext
```
Comment (manuelvicnt) [inline app/src/main/java/com/example/android/roomwordssample/WordViewModel.kt:39]: `viewModelScope.coroutineContext` is not really needed. This is needed if you want the flow to be cancelled as soon as the ViewModel is destroyed. But it's going to do it anyway after the timeout. Remove if you want to simplify the code, leave it if it makes more sense to you.

**Review by manuelvicnt — APPROVED** [review]
Added two nits. Feel free to merge after those are addressed :)

**Review by manuelvicnt — APPROVED** [review]

**Review by manuelvicnt — APPROVED** [review]

**Pull request merged**